pytorch_lightning/trainer/connectors/debugging_connector.py

```diff
@@ -31,16 +31,34 @@ def on_init_start(
             overfit_batches,
             fast_dev_run
     ):
+        if not isinstance(fast_dev_run, (bool, int)):
+            raise MisconfigurationException(
+                f'fast_dev_run={fast_dev_run} is not a valid configuration.'
+                ' It should be either a bool or an int >= 0'
+            )
+
+        if isinstance(fast_dev_run, int) and (fast_dev_run < 0):
+            raise MisconfigurationException(
+                f'fast_dev_run={fast_dev_run} is not a'
+                ' valid configuration. It should be >= 0.'
+            )
 
         self.trainer.fast_dev_run = fast_dev_run
-        if self.trainer.fast_dev_run:
-            limit_train_batches = 1
-            limit_val_batches = 1
-            limit_test_batches = 1
+        fast_dev_run = int(fast_dev_run)
+
+        # set fast_dev_run=True when it is 1, used while logging
+        if fast_dev_run == 1:
+            self.trainer.fast_dev_run = True
```

Comment:
The reason I did this to make it follow `log_every_n_steps` when `fast_dev_run >1` as Thomas suggested. If I just do `if self.fast_dev_run` below it will log at every step, so I made it a strict conditional with `if self.fast_dev_run is True` and made the changes here accordingly.

Reply:
When fast dev run is set, we prefer not to have logging turned on at all. fast dev run is there to quickly debug crashing code in any stage of training, and so polluting the logs with these runs is not desired.

Reply:
on the master, it does log in case of `fast_dev_run=True`. You want me to change that?? @awaelchli cc: @tchaton

Reply:
oh, that's unfortunate :( Then we may need to follow up on this in a new PR / issue because I am fairly confident that this is not intended behaviour. The logic to differentiate between True and 1 is a bit unsatisfiying in my view.
Thanks for clarification, I'll remove the "request for changes"

Reply:
You might be wanting to test elaborate logging calls though (this was a use case for me on a previous project)